subsys/bluetooth/host/hci_core.c

```diff
@@ -6692,3 +6713,20 @@ int bt_le_oob_get_sc_data(struct bt_conn *conn,
 	return bt_smp_le_oob_get_sc_data(conn, oobd_local, oobd_remote);
 }
 #endif
+
+#if defined(CONFIG_BT_PRIVACY)
+int bt_set_rpa_timeout(u16_t timeout)
+{
+	int err;
+
+	/* Restart timer even with new value */
+	k_delayed_work_cancel(&bt_dev.rpa_update);
+	k_delayed_work_submit(&bt_dev.rpa_update,
+			      K_SECONDS(rpa_timeout_seconds));
+
+	rpa_timeout_seconds = timeout;
+	err = bt_hci_set_rpa_timeout(rpa_timeout_seconds);
+
+	return err;
+}
+#endif /* CONFIG_BT_PRIVACY */
```

Comment:
Regarding bt_set_rpa_timeout: I could move this to vendor specific interface, if it doesn't fit in...